sematic/user_settings.py

```diff
@@ -88,57 +46,326 @@ class SettingsVar(enum.Enum):
+                for var, value in self.profiles[profile].items()
+            }
+
+    def get_active_profile_settings(self) -> ProfileSettingsType:
+        """
+        Returns the settings associated with the active profile.
+        """
+        return self.profiles[self.active_profile]
+
+    def get_inactive_profiles(self) -> List[str]:
+        """
+        Returns a list of existing profile names other than the active one.
+        """
+        return [
+            profile
+            for profile in self.profiles.keys()
+            if profile != self.active_profile
+        ]
+
+    def set(self, var: SettingsVar, value: Optional[str]) -> None:
+        """
+        Sets the specified value for the specified variable, for the active profile.
+
+        Deletes the entry if the value is None.
+        """
+        active_profile_settings = self.get_active_profile_settings()
+
+        if value is not None:
+            active_profile_settings[var] = value
+            return
+
+        if var not in active_profile_settings:
+            raise ValueError(f"{var.value} is not present in the active profile!")
+
+        del active_profile_settings[var]
+
+    def set_profile(self, profile: str) -> None:
+        """
+        Sets the currently active profile.
+        """
+        if len(profile) == 0:
+            raise ValueError("The profile name cannot be empty!")
+
+        self.active_profile = profile
+
+        if self.active_profile not in self.profiles:
+            self.profiles[self.active_profile] = {}
+
+    def delete_profile(self, profile: str) -> None:
+        """
+        Deletes the specified profile.
+        """
+        if len(profile) == 0:
+            raise ValueError("The profile name cannot be empty!")
+
+        if profile == self.active_profile:
+            raise ValueError(
+                "Cannot delete the active profile! Switch to another profile first!"
             )
+
+        if profile not in self.profiles:
+            raise ValueError(f"Profile '{profile}' does not exist!")
+
+        del self.profiles[profile]
+
+
+# global settings cache
+_settings: Optional[UserSettings] = None
+
+
+def _get_settings_file() -> str:
+    """
+    Returns the path to the settings file according to the user configuration.
+    """
+    return os.path.join(get_config_dir(), _SETTINGS_FILE)
+
+
+def _load_settings() -> UserSettings:
+    """
+    Loads the settings from the configured settings file.
+    """
+    try:
+        with open(_get_settings_file(), "r") as f:
+            raw_settings = yaml.load(f, yaml.Loader)
+
+    except FileNotFoundError:
+        logger.debug("Settings file %s not found", _get_settings_file())
+        return get_default_settings()
+
+    if raw_settings is None:
+        return get_default_settings()
+
+    try:
+        return UserSettings(**raw_settings)
+    except TypeError:
+        # the settings file has an older syntax and should be corrected
```

Comment:
Instead of this, how about we put a `settings_schema_version` config in the file. If it is not present, assume the schema version is 0. The current schema can be 1.